controllers/drpolicy_controller.go

```diff
@@ -378,5 +387,67 @@ func (u *objectUpdater) finalizerRemove() error {
 func (r *DRPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ramen.DRPolicy{}).
+		Watches(
+			&source.Kind{Type: &corev1.ConfigMap{}},
+			handler.EnqueueRequestsFromMapFunc(r.configMapMapFunc),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
+			handler.EnqueueRequestsFromMapFunc(r.secretMapFunc),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		).
 		Complete(r)
 }
+
+func (r *DRPolicyReconciler) configMapMapFunc(configMap client.Object) []reconcile.Request {
+	if configMap.GetName() != HubOperatorConfigMapName || configMap.GetNamespace() != NamespaceName() {
+		return []reconcile.Request{}
+	}
+
+	drpolicies := &ramen.DRPolicyList{}
+	if err := r.Client.List(context.TODO(), drpolicies); err != nil {
+		return []reconcile.Request{}
+	}
+
+	requests := make([]reconcile.Request, len(drpolicies.Items))
+	for i, drpolicy := range drpolicies.Items {
+		requests[i].Name = drpolicy.GetName()
+	}
+
+	return requests
+}
+
+func (r *DRPolicyReconciler) secretMapFunc(secret client.Object) (requests []reconcile.Request) {
+	drpolicies := &ramen.DRPolicyList{}
```

Comment:
(future) We should also filter out secrets within our namespace, external namespace secrets are not supported. Although current CRD does take a namespace in addition to the name of the secret, which needs to change.

Reply:
Yes, I suppose an issue tracker is needed to remove the namespace input for a secret reference.